task/net/src/bsp/sidecar_1.rs

```diff
@@ -0,0 +1,244 @@
+    None,
+    Ksz8463Status { port: u8, status: u16 },
+    Vsc8552Status { port: u8, status: u16 },
+}
+ringbuf!(Trace, 16, Trace::None);
+
+////////////////////////////////////////////////////////////////////////////////
+
+pub struct Bsp {
+    ksz: Ksz8463,
+}
+
+impl Bsp {
+    pub fn new() -> Self {
+        let spi = Spi::from(SPI.get_task_id()).device(KSZ8463_SPI_DEVICE);
+        let ksz = Ksz8463::new(spi, gpio_api::Port::A.pin(0), false);
+
+        Self { ksz }
+    }
+
+    pub fn configure_ethernet_pins(&self) {
+        // This board's mapping:
+        //
+        // RMII REF CLK     PA1
+        // RMII RX DV       PA7
+        //
+        // RMII RXD0        PC4
+        // RMII RXD1        PC5
+        //
+        // RMII TX EN       PG11
+        // RMII TXD1        PG12
+        // RMII TXD0        PG13
+        //
+        // MDIO             PA2
+        //
+        // MDC              PC1
+        //
+        // (it's _almost_ identical to the STM32H7 Nucleo, except that
+        //  TXD1 is on a different pin)
+        //
+        //  The MDIO/MDC lines run at Speed::Low because otherwise the VSC8504
+        //  refuses to talk.
+        use gpio_api::*;
+        let gpio = Gpio::from(GPIO.get_task_id());
+        let eth_af = Alternate::AF11;
+
+        // RMII
+        gpio.configure(
+            Port::A,
+            (1 << 1) | (1 << 7),
+            Mode::Alternate,
+            OutputType::PushPull,
+            Speed::VeryHigh,
+            Pull::None,
+            eth_af,
+        )
+        .unwrap();
+        gpio.configure(
+            Port::C,
+            (1 << 4) | (1 << 5),
+            Mode::Alternate,
+            OutputType::PushPull,
+            Speed::VeryHigh,
+            Pull::None,
+            eth_af,
+        )
+        .unwrap();
+        gpio.configure(
+            Port::G,
+            (1 << 11) | (1 << 12) | (1 << 13),
+            Mode::Alternate,
+            OutputType::PushPull,
+            Speed::VeryHigh,
+            Pull::None,
+            eth_af,
+        )
+        .unwrap();
+
+        // SMI (MDC and MDIO)
+        gpio.configure(
+            Port::A,
+            1 << 2,
+            Mode::Alternate,
+            OutputType::PushPull,
+            Speed::Low,
+            Pull::None,
+            eth_af,
+        )
+        .unwrap();
+        gpio.configure(
+            Port::C,
+            1 << 1,
+            Mode::Alternate,
+            OutputType::PushPull,
+            Speed::Low,
+            Pull::None,
+            eth_af,
+        )
+        .unwrap();
+    }
+
+    pub fn configure_phy(&self, eth: &mut eth::Ethernet) {
+        // The KSZ8463 connects to the SP over RMII, then sends data to the
+        // VSC8552 over 100-BASE FX
+        self.ksz.configure();
+
+        // The VSC8552 connects the KSZ switch to the management network
+        // over SGMII
+        configure_vsc8552(eth);
+    }
+
+    pub fn wake(&self, eth: &mut eth::Ethernet) {
```

Comment:
So...what's this for?

Reply:
Moved out of the Idol API, added a new `Option<u64>` to ask the `net` task to run periodic logging at a BSP-specified interval.